### PR TITLE
New docker implementation 

### PR DESCRIPTION
**Pull request by themajerr**

### CODE REVIEW

File: backend/.env [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # PORT=8000
-# HOST=0.0.0.0
+HOST=0.0.0.0
```
Comment (radek2s) [inline backend/.env:2]: This should be commented and ENV should be added to `backend/Dockerfile`

```
WORKDIR /app/backend
ENV HOST=0.0.0.0
```

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,10 @@
...
 WORKDIR /app
-COPY package.json .
-RUN npm install 
+COPY ./package*.json ./
+RUN npm install
 COPY . ./
-EXPOSE 3000
+RUN npm run build
+RUN npm install -g serve
 EXPOSE 8000
```
Comment (radek2s) [inline Dockerfile:8]: This `EXPOSE 8000` is redundant

File: backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM node:18-alpine
+WORKDIR /app/backend
+COPY /package*.json ./
+RUN npm install
+COPY . ./
+EXPOSE 8000
```
Comment (radek2s) [inline backend/Dockerfile:6]: When you have `.dockerignore` file you don't have to use copy method twice. Dockerfile will define files to exclude. So copy onece and then after `npm install` you will have all required dependencies to start an application

**Review by radek2s — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+version: "3.9"
+services:
+  front_end:
```
[the remark this answers is not in the document]
Comment (themajerr) [reply docker-compose.yml]: Corrected

File: backend/.env (hunk 1 above)
Comment (themajerr) [reply backend/.env]: Corrected

File: starting_script.sh [hunk 5]
```diff
@@ -1,4 +0,0 @@
-#!/bin/sh
```
[the remark this answers is not in the document]
Comment (themajerr) [reply starting_script.sh:1]: Done for both parts (front and back)

File: backend/Dockerfile (hunk 3 above)
Comment (themajerr) [reply backend/Dockerfile]: Corrected, now both files copy only once

**Review by radek2s — APPROVED** [review]

**Pull request merged**